components/support/migration/src/main/java/mozilla/components/support/migration/MigrationIntentProcessor.kt

```diff
@@ -0,0 +1,48 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.support.migration
+
+import android.content.Intent
+import android.content.Intent.ACTION_VIEW
+import mozilla.components.feature.intent.processing.IntentProcessor
+import mozilla.components.support.migration.state.MigrationProgress
+import mozilla.components.support.migration.state.MigrationStore
+import mozilla.components.support.utils.toSafeIntent
+
+/**
+ * An [IntentProcessor] that checks if we're in a migration state.
+ *
+ * ⚠️ When using this processor, ensure this is the first processor to be invoked if there are multiple.
+ */
+class MigrationIntentProcessor(private val store: MigrationStore) : IntentProcessor {
+
+    /**
+     * Matches itself with [ACTION_VIEW] intents in order to ensure processing all of incoming intents.
+     */
+    override fun matches(intent: Intent): Boolean = intent.toSafeIntent().action == ACTION_VIEW
```

Comment:
Wouldn't we want to handle other intents like `ACTION_SEND` here too?

Reply:
I realized that we can just return true instead while migrating so I'm going to change this to that instead. :)

Reply:
For reference, this is the change, so we end up catching all intents. I didn't need to do any fancy in the end:
```diff
-override fun matches(intent: Intent): Boolean = intent.toSafeIntent().action == ACTION_VIEW
+override fun matches(intent: Intent): Boolean = store.state.progress == MigrationProgress.MIGRATING
```